spot-contracts/contracts/FeePolicy.sol

```diff
@@ -0,0 +1,259 @@
+    ///        as a fixed point number with {DECIMALS} places.
+    function updatePerpBurnFees(uint256 maxPerpBurnFeePerc_) external onlyOwner {
+        require(maxPerpBurnFeePerc_ <= ONE, "FeeStrategy: perc too high");
+        maxPerpBurnFeePerc = maxPerpBurnFeePerc_;
+    }
+
+    /// @notice Updates the vault mint fee parameters.
+    /// @param maxVaultMintFeePerc_ The new vault mint fee ceiling percentage
+    ///        as a fixed point number with {DECIMALS} places.
+    function updateVaultMintFees(uint256 maxVaultMintFeePerc_) external onlyOwner {
+        require(maxVaultMintFeePerc_ <= ONE, "FeeStrategy: perc too high");
+        maxVaultMintFeePerc = maxVaultMintFeePerc_;
+    }
+
+    /// @notice Updates the vault burn fee parameters.
+    /// @param maxVaultBurnFeePerc_ The new vault burn fee ceiling percentage
+    ///        as a fixed point number with {DECIMALS} places.
+    function updateVaultBurnFees(uint256 maxVaultBurnFeePerc_) external onlyOwner {
+        require(maxVaultBurnFeePerc_ <= ONE, "FeeStrategy: perc too high");
+        maxVaultBurnFeePerc = maxVaultBurnFeePerc_;
+    }
+
+    /// @notice Update the parameters determining the slope and asymptotes of the sigmoid fee curve.
+    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {DECIMALS} places.
+    function updateRolloverFees(SigmoidParams calldata p) external onlyOwner {
+        require(p.lower >= -int256(SIGMOID_BOUND), "FeeStrategy: sigmoid lower bound too low");
+        require(p.upper <= int256(SIGMOID_BOUND), "FeeStrategy: sigmoid upper bound too high");
+        require(p.lower <= p.upper, "FeeStrategy: sigmoid asymptotes invalid");
+        perpRolloverFeeAPR.lower = p.lower;
+        perpRolloverFeeAPR.upper = p.upper;
+        perpRolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpMintFeePerc() external override returns (uint256) {
+        return isOverSubscribed() ? 0 : maxPerpMintFeePerc;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpBurnFeePerc() external override returns (uint256) {
+        return isOverSubscribed() ? maxPerpBurnFeePerc : 0;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpRolloverFeePerc() external override returns (int256) {
+        IBondIssuer bondIssuer = perp.bondIssuer();
+        SubscriptionState memory s = computeSubscriptionState(bondIssuer);
+
+        int256 rolloverAPR = Sigmoid.compute(
+            s.normalizedSubscriptionRatio.toInt256(),
+            perpRolloverFeeAPR.lower,
+            perpRolloverFeeAPR.upper,
+            perpRolloverFeeAPR.growth,
+            ONE.toInt256()
+        );
+
+        // We calculate the rollover fee for the given cycle by dividing the annualized rate
+        // by the number of cycles in any given year.
+        return (rolloverAPR * bondIssuer.maxMaturityDuration().toInt256()) / ONE_YEAR_SEC;
```

Comment:
This should be issue frequency, not duration, right?

Dividing by 52 vs dividing by 12 is a big difference

Reply:
It should be divided by 12 as we rollover the entire supply only 12 times

Reply:
Makes sense. One option maybe worth considering--We could save some gas by not doing any normalization here in code.

I.e. We configure the curve then use it's output directly. But we configure such that the end result is a "yearly rate".

This would save a division, an external call, and a couple reads at the least.

Reply:
Right. That would save the extra query. We just have to be extra cautious while setting the parameters ..